UpdatedImageProcessing/UpdatedImageProcessing/TargetDetection/background_color_nullifier.py

```diff
@@ -87,108 +137,80 @@ def nullify_color_and_recrop_target(captured_image, color_difference_threshold):
-                        stopping_x.append(x)
-                        break
-                    else:
-                        pixel_access_recaptured_image[x, y] = (255, 255, 255, 0)
-                    x -= 1
 
-            x_min = min(stopping_x)
-            y_min = min(stopping_y)
-            x_max = max(stopping_x)
-            y_max = max(stopping_y)
+            index = 0
+            while (index < 5):
+                pixel_access_recaptured_image = recaptured_image.load()
+                increased_empty_space_area = 0
+                for x in range(recaptured_image.width):
+                    for y in range(recaptured_image.height):
+                        number_of_blanks_around_1 = 0
+                        number_of_blanks_around_2 = 0
+
+                        if (pixel_access_recaptured_image[x, y][3] != 0):
+                            try:
+                                if (pixel_access_recaptured_image[x - 1, y][3] == 0):
+                                    number_of_blanks_around_1 += 1
+                            except:
+                                pass
+
+                            try:
+                                if (pixel_access_recaptured_image[x + 1, y][3] == 0):
+                                    number_of_blanks_around_1 += 1
+                            except:
+                                pass
+
+                            try:
+                                if (pixel_access_recaptured_image[x, y - 1][3] == 0):
+                                    number_of_blanks_around_1 += 1
+                            except:
+                                pass
+
+                            try:
+                                if (pixel_access_recaptured_image[x, y + 1][3] == 0):
+                                    number_of_blanks_around_1 += 1
+                            except:
+                                pass
+
+
+                            try:
+                                if (pixel_access_recaptured_image[x - 2, y][3] == 0):
+                                    number_of_blanks_around_2 += 1
+                            except:
+                                pass
+
+                            try:
+                                if (pixel_access_recaptured_image[x + 2, y][3] == 0):
+                                    number_of_blanks_around_2 += 1
+                            except:
+                                pass
+
+                            try:
+                                if (pixel_access_recaptured_image[x, y - 2][3] == 0):
+                                    number_of_blanks_around_2 += 1
+                            except:
+                                pass
+
+                            try:
+                                if (pixel_access_recaptured_image[x, y + 2][3] == 0):
+                                    number_of_blanks_around_2 += 1
+                            except:
+                                pass
```

Comment:
From line 156 to line 204 here, this seems like this can be done more concisely with either a for or while loop.

Reply:
Or separating out the logic to use multiple methods.